manim/utils/tex.py

```diff
@@ -1,298 +1,179 @@
+            if tex_compiler is not None
+            else TexTemplate.default_tex_compiler
         )
-
-    def get_text_for_tex_mode(self, expression):
-        """Inserts an expression within an ``align*`` environment into the TeX template.
-
-        Parameters
-        ----------
-        expression : :class:`str`
-            The string containing the (math) expression to be typeset,
-            e.g. ``$\\sqrt{2}$``
-
-        Returns
-        -------
-        :class:`str`
-            LaTeX code based on template, containing the given expression and ready for typesetting
-        """
-        return self.get_text_for_env("align*", expression)
-
-
-class TexTemplate(TexTemplateFromFile):
-    """
-    Class for dynamically managing a TeX template
-    """  # TODO: Add attributes (when dataclasses are implemented)
-
-    CONFIG = {
-        "documentclass": ["standalone", ["preview"]],
-        "common_packages": [
-            ["babel", ["english"]],
-            "amsmath",
-            "amssymb",
-            "dsfont",
-            "setspace",
-            "tipa",
-            "relsize",
-            "textcomp",
-            "mathrsfs",
-            "calligra",
-            "wasysym",
-            "ragged2e",
-            "physics",
-            "xcolor",
-            "microtype",
-        ],
-        "tex_packages": [["inputenc", ["utf8"]], ["fontenc", ["T1"]]],
-        "ctex_packages": [["ctex", ["UTF8"]]],
-        "common_preamble_text": r"\linespread{1}" "\n",
-        "tex_preamble_text": r"\DisableLigatures{encoding = *, family = *}" "\n",
-        "ctex_preamble_text": "",
-        "document_prefix": "",
-        "document_suffix": "",
-    }
-
-    def __init__(self, **kwargs):
-        digest_config(self, kwargs)
-        self.rebuild_cache()
-
-    def rebuild_cache(self):
-        """For faster access, the LaTeX template's code is cached.
-        If the base file is modified, the cache needs to be rebuilt."""
-        tpl = self.generate_tex_command(
-            "documentclass",
-            required_params=[self.documentclass[0]],
-            optional_params=self.documentclass[1],
+        self.output_format = (
+            output_format
+            if output_format is not None
+            else TexTemplate.default_output_format
+        )
+        self.documentclass = (
+            documentclass
+            if documentclass is not None
+            else TexTemplate.default_documentclass
+        )
+        self.preamble = (
+            preamble if preamble is not None else TexTemplate.default_preamble
+        )
+        self.placeholder_text = (
+            placeholder_text
+            if placeholder_text is not None
+            else TexTemplate.default_placeholder_text
+        )
+        self.post_doc_commands = (
+            post_doc_commands
+            if post_doc_commands is not None
+            else TexTemplate.default_post_doc_commands
+        )
+        self.rebuild()
+
+    def rebuild(self):
```

Comment:
no docs?

Reply:
If you want to use a template, you mightcreate a TexTemplate() and you might add_to_preamble(), but you don't need to call rebuild().

We have to decide at some point: who is the documentation for?
Are we documenting for people who want to use manim or for people who want to hack manim? The two are not the same, but, by nature of this project, they are not clearly separated either.

This function is only useful for people who want to hack away at the template system, not for people who are using manim.

Reply:
Ideally, documentation could be used by either users or developers. I do agree that user-docs should take precedence over dev-docs.

However, with the current state of things, *any* docs are better than none.

Reply:
or rename it to `_rebuild()` and just 'document' it with a # comment ?

Reply:
Yes for making it private, but even private functions deserve a docstring. Whatever you would put in a comment, put it in a docstring.

Reply:
ok. I'll do that quickly. hang on.